pkg/synopsysctl/cmd_update.go

```diff
@@ -1216,56 +1215,56 @@ var updatePolarisCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		polarisSecret, err := kubeClient.CoreV1().Secrets(namespace).Get("polaris", metav1.GetOptions{})
+		p, err := getPolarisFromSecret()
 		if err != nil {
 			return err
 		}
 
-		polarisSecretBytes, ok := polarisSecret.Data["polaris"]
-		if !ok {
-			return fmt.Errorf("polaris entry is missing in the secret")
-		}
-
-		var p *polaris.Polaris
-		if err := json.Unmarshal(polarisSecretBytes, &p); err != nil {
+		newPolaris, err := updatePolaris(*p, cmd.Flags())
+		if err != nil {
 			return err
 		}
 
-		currPolaris, err := updatePolaris(*p, cmd.Flags())
+		file, err := polarisToTmpComponentFile(*newPolaris)
 		if err != nil {
 			return err
 		}
 
-		// Marshal Polaris
-		polarisByte, err := json.Marshal(currPolaris)
+		defer os.RemoveAll(filepath.Dir(file))
+
+		fmt.Print("Updating Polaris")
+		ch := make(chan struct{})
+		go printWaitingDots(time.Second*5, ch)
+
+		out, err := RunKubeCmd(restconfig, kubeClient, "apply", "--validate=false", "-f", file)
 		if err != nil {
-			return err
+			close(ch)
+			return fmt.Errorf("Couldn't update polaris |  %+v - %s", out, err)
 		}
 
-		polarisSecret.Data["polaris"] = polarisByte
-
-		polarisSecret, err = kubeClient.CoreV1().Secrets(polarisSecret.Namespace).Update(polarisSecret)
+		// Marshal Polaris
+		polarisByte, err := json.Marshal(newPolaris)
 		if err != nil {
 			return err
 		}
 
-		myscheme := runtime.NewScheme()
-		_ = clientgoscheme.AddToScheme(myscheme)
-
-		// create controller runtime client
-		cl, _ := runtimeclient.New(restconfig, runtimeclient.Options{
-			Scheme: myscheme,
-		})
+		polarisSecret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "polaris",
+				Namespace: namespace,
+			},
+			Data: map[string][]byte{
+				"polaris": polarisByte,
+			},
+		}
 
-		// Start reconcile process
-		if _, err := flying_dutchman.MetaReconcile(polarisSecret, &polaris.PolarisReconciler{
-			Client:      cl,
-			Scheme:      myscheme,
-			IsDryRun:    false,
-			IsOpenShift: false,
-		}); err != nil {
+		_, err = kubeClient.CoreV1().Secrets(namespace).Update(polarisSecret)
+		if err != nil {
+			close(ch)
 			return err
 		}
+		close(ch)
+		fmt.Println("\nPolaris has been successfully updated!")
```

Comment:
move all fmt to logger?

Reply:
Using the logger would be ugly for displaying such messages.